File: course-evaluation-system/backend/src/main/java/com/must/courseevaluation/service/CourseService.java
package com.must.courseevaluation.service;

import com.must.courseevaluation.dto.CourseDto;
import com.must.courseevaluation.dto.CourseScheduleDto;
import com.must.courseevaluation.model.Course;

import java.util.List;

public interface CourseService {
    List<CourseDto> getAllCourses();
    CourseDto getCourseById(Long id);
    List<CourseDto> getCoursesByFaculty(Long facultyId);
    List<CourseDto> getCoursesByTeacher(Long teacherId);
    List<CourseDto> getCoursesByType(Course.CourseType type);
    List<CourseDto> searchCourses(String keyword);
    CourseDto createCourse(CourseDto courseDto);
    CourseDto updateCourse(CourseDto courseDto);
    void deleteCourse(Long id);
    
    // 课程时间表相关方法
    /**
     * 获取课程的时间安排
     */
    List<CourseScheduleDto> getCourseSchedules(Long courseId);
    
    /**
     * 添加课程时间安排
     */
    CourseScheduleDto addCourseSchedule(Long courseId, CourseScheduleDto scheduleDto);
    
    /**
     * 更新课程时间安排
     */
    CourseScheduleDto updateCourseSchedule(Long scheduleId, CourseScheduleDto scheduleDto);
    
    /**
     * 删除课程时间安排
     */
    void deleteCourseSchedule(Long scheduleId);
    
    /**
     * 批量设置课程时间安排（先删除再添加）
     */
    List<CourseScheduleDto> setCourseSchedules(Long courseId, List<CourseScheduleDto> scheduleDtos);
    
    /**
     * 根据时间段查找课程（用于AI推荐时排除冲突）
     */
    List<CourseDto> findCoursesBySchedule(Integer dayOfWeek, Integer timePeriod);
    
    /**
     * 查找与用户时间表不冲突的课程
     */
    List<CourseDto> findCoursesWithoutConflict(Long userId);
} 

File: course-evaluation-system/frontend/src/services/course.service.js
import http from './http.service'

class CourseService {
  getAllCourses() {
    return http.get('courses')
      .catch(error => {
        // 如果是401错误(未授权)，可能是token问题，尝试直接请求
        if (error.response && error.response.status === 401) {
          console.log('课程列表获取失败，尝试作为公共API重试')
          // 使用不带token的请求重试
          return http.get('courses', { 
            headers: { 'Skip-Auth': true } // 自定义标记，在拦截器中可以处理
          })
        }
        return Promise.reject(error)
      })
  }
  
  getCourseById(id) {
    return http.get(`courses/${id}`)
      .catch(error => {
        if (error.response && error.response.status === 401) {
          console.log('课程详情获取失败，尝试作为公共API重试')
          return http.get(`courses/${id}`, { 
            headers: { 'Skip-Auth': true }
          })
        }
        return Promise.reject(error)
      })
  }
  
  getCoursesByFaculty(facultyId) {
    return http.get(`courses/faculty/${facultyId}`)
      .catch(error => {
        if (error.response && error.response.status === 401) {
          console.log('课程(按院系)获取失败，尝试作为公共API重试')
          return http.get(`courses/faculty/${facultyId}`, { 
            headers: { 'Skip-Auth': true }
          })
        }
        return Promise.reject(error)
      })
  }
  
  getCoursesByTeacher(teacherId) {
    return http.get(`courses/teacher/${teacherId}`)
      .catch(error => {
        if (error.response && error.response.status === 401) {
          console.log('课程(按教师)获取失败，尝试作为公共API重试')
          return http.get(`courses/teacher/${teacherId}`, { 
            headers: { 'Skip-Auth': true }
          })
        }
        return Promise.reject(error)
      })
  }
  
  getCoursesByType(type) {
    return http.get(`courses/type/${type}`)
      .catch(error => {
        if (error.response && error.response.status === 401) {
          console.log('课程(按类型)获取失败，尝试作为公共API重试')
          return http.get(`courses/type/${type}`, { 
            headers: { 'Skip-Auth': true }
          })
        }
        return Promise.reject(error)
      })
  }
  
  searchCourses(keyword) {
    return http.get(`courses/search?keyword=${keyword}`)
      .catch(error => {
        if (error.response && error.response.status === 401) {
          console.log('课程搜索失败，尝试作为公共API重试')
          return http.get(`courses/search?keyword=${keyword}`, { 
            headers: { 'Skip-Auth': true }
          })
        }
        return Promise.reject(error)
      })
  }
  
  createCourse(data) {
    return http.post('courses', data)
  }
  
  updateCourse(id, data) {
    return http.put(`courses/${id}`, data)
  }
  
  deleteCourse(id) {
    return http.delete(`courses/${id}`)
  }
  
  // 获取课程评价统计
  getCourseRatings(id) {
    return http.get(`courses/${id}/ratings`)
      .catch(error => {
        if (error.response && error.response.status === 401) {
          return http.get(`courses/${id}/ratings`, { 
            headers: { 'Skip-Auth': true }
          })
        }
        return Promise.reject(error)
      })
  }
}

export default new CourseService() 